### PR TITLE
Add a linear eachindex definition

### LINKED ISSUE #64: eachindex
Following the discussion at JuliaLang/julia#10858, I realized that `eachindex` is not implemented in Compat for julia 0.3.  Might it make sense to implement it?

### PR DESCRIPTION
**Pull request by mbauman**
This is my first Compat addition - so it'd be good if someone could double-check my version number and such.

Fixes #64.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -217,3 +217,17 @@ if VERSION < v"0.4.0-dev+4319"
     @test @compat Tuple{:a, Tuple{:b}} == (:a, (:b,))
     @test @compat Tuple{:a, Tuple{:b, :c}} == (:a, (:b, :c))
 end
+
+# Ensure eachindex iterates over the whole array
+A = reshape(1:20,4,5)
```
Comment (kmsquire) [inline test/runtests.jl]: Conflicts with https://github.com/JuliaLang/Compat.jl/blob/d49e234ddb563b558d890a5cfbd87dc6f641c56f/test/runtests.jl#L204

File: test/runtests.jl [hunk 2]
```diff
@@ -217,3 +217,19 @@ if VERSION < v"0.4.0-dev+4319"
     @test @compat Tuple{:a, Tuple{:b}} == (:a, (:b,))
     @test @compat Tuple{:a, Tuple{:b, :c}} == (:a, (:b, :c))
 end
+
+# Ensure eachindex iterates over the whole array
+let
+    AA = reshape(1:20,4,5)
```
Comment (kmsquire) [inline test/runtests.jl]: Note that using let blocks in this way prevents `AA` (and other vars) from leaking out, but doesn't prevent inward leakage (e.g., try defining `type AA end` and running this block).

Comment (mbauman) [inline test/runtests.jl]: Yeah, that's why it's now `AA` instead of `A`.  I just figured it'd be best practice in any case.

Comment (kmsquire) [inline test/runtests.jl]: Okay.  Having been bit by predefined variables before (in `Base`'s tests), I'm wondering if it isn't better to explicitly make everything a `let` variable:

```julia
let AA, s, B
    AA = reshape(1:20, 4, 5)
    ...
```

Comment (mbauman) [inline test/runtests.jl]: Ah, interesting.  That would prevent the collision with a `type AA`, wouldn't it?

Comment (mbauman) [inline test/runtests.jl]: Very cool.  Thanks for the review and the tip! :+1:

**Comment by timholy** [comment]
Version number looks right. Thanks!

**Comment by tkelman** [comment]
probably worth a line in the readme

**Comment by timholy** [comment]
Done.

**Comment by mbauman** [comment]
Thanks!

**Pull request merged**